Add metrics collection and metrics-based dashboards info to Redis Otel (#4286)
SumoLogic/sumologic-documentation: docs/integrations/databases/opentelemetry/redis-opentelemetry.md

## Log and metrics types

This app supports metrics and logs for Redis in Cluster mode or Standalone mode. The Redis logs are generated in files as configured in the cluster node configuration files in `/etc/redis/redis.conf`. For more details on Redis logs, see [Redis documentation](https://redislabs.com/ebook/part-2-core-concepts/chapter-5-using-redis-for-application-support/5-1-logging-to-redis/).

The app supports metrics generated by the [Redis Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/redisreceiver/documentation.md).

## Creating fields in Sumo Logic for Redis

The following are [fields](/docs/manage/fields/) that will be created as part of the Redis App install if not already present.

* **`db.cluster.name`**. User configured. Enter a name to identify this Redis cluster. This cluster name will be shown in the Sumo Logic dashboards.
* **`db.system`**. Has fixed value of redis.
* **`deployment.environment`**. User configured. This is the deployment environment where the Redis cluster resides. For example: dev, prod or qa.

> * **`db.node.name`**. Has the value of host name of the machine which is being monitored

Reviewer: ```suggestion
* **`db.node.name`**. Has the value of host name of the machine which is being monitored.
```

Author: fixed it.

* **`sumo.datasource`**. Has fixed value of redis.

## Prerequisites

### For metrics collection

> The Redis INFO command returns information and statistics about a Redis server (see Redis [INFO](https://redis.io/commands/info) for details). The Redis receiver extracts values from the result and converts them to open telemetry metrics.

Reviewer: ```suggestion
The Redis INFO command returns information and statistics about a Redis server (see Redis [INFO](https://redis.io/commands/info) for details). The Redis receiver extracts values from the result and converts them to OpenTelemetry metrics.
```

Author: fixed it.

* **`db.cluster.name`**. User configured. Enter a name to identify this Redis cluster. This cluster name will be shown in the Sumo Logic dashboards.
* **`db.system`**. Has fixed value of redis.
* **`deployment.environment`**. User configured. This is the deployment environment where the Redis cluster resides. For example: dev, prod or qa.

> * **`webengine.node.name`**. Has the value of host name of the machine which is being monitored

Reviewer 2: this should be db.node.name ?

Author: fixed it

In this step, you will configure the yaml file required for Redis Collection.

Below are the inputs required:

- **`endpoint (no default)`**. The hostname and port of the Redis instance, separated by a colon. (For example: `localhost:6379`.)
- **Redis logs Path**. Enter the path to the log file for your Redis instance.

The log file path configured to capture redis logs must be given here. The files are typically located in `/var/log/redis/redis-server.log`. If you are using a customized path, check the [`redis.conf`](https://download.redis.io/redis-stable/redis.conf) file for this information.

You can add any custom fields which you want to tag along with the data ingested in Sumo. Click on the **Download YAML File** button to get the yaml file.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/Redis-YAML.png' style={{border:'1px solid gray'}} alt="Configuration" />

:::note

>   Each Redis replica requires an independent sumo opentelemetry collector to be installed for effective log and metric monitoring.

Reviewer: ```suggestion
  Each Redis replica requires an independent Sumo Logic OpenTelemetry collector to be installed for effective log and metric monitoring.
```

Author: fixed it.

In this step, you will configure the yaml file required for Redis Collection.

Below are the inputs required:

> - **`endpoint (no default)`**. The hostname and port of the Redis instance, separated by a colon..

Reviewer 2: Can you give an example for clarity here

Author: fixed it.

Use this dashboard to review keyspace hit, Cache miss rate by host, CPU - System vs User usage by host, and Network Usage - Inbound/Outbound by Host.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/Redis-Cluster-Operations.png' alt="Redis Cluster Operations dashboards" />

### Cluster Overview

The **Redis - Cluster Overview** dashboard provides a high-level view of cluster infrastructure, comparisons, and trends.

Use this dashboard to review Average Master Uptime, Connected Slaves, Redis Uptime by Host, Commands by Host, and Instantaneous OPS per sec by Host.

<img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/Redis-Cluster-Overview.png' alt="Redis Cluster Overview dashboards" />

### Command Stats Delta

The **Redis - Command Stats Delta** dashboard provides an at-a-glance view of commands running on your Redis server.

Use this dashboard to review String Commands, Key Commands, Hash Commands, Sets Commands, Sorted Sets Commands, List Commands, Transaction Commands, and Scripting Commands.

> <img src='https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Redis-OpenTelemetry/Redis-Command-Stats-Delta.png' alt="Redis Command Stats Delta dashboards" />

Reviewer 3: This image is not rendering in my local build. Can you verify that the URL is correct?
